website/server/libs/payments.js

```diff
@@ -321,4 +322,180 @@ api.buyGems = async function buyGems (data) {
   await data.user.save();
 };
 
+/**
+ * Allows for purchasing a user subscription, group subscription or gems with Stripe
+ *
+ * @param  options
+ * @param  options.token  The stripe token generated on the front end
+ * @param  options.user  The user object who is purchasing
+ * @param  options.gift  The gift details if any
+ * @param  options.sub  The subscription data to purchase
+ * @param  options.groupId  The id of the group purchasing a subscription
+ * @param  options.email  The email enter by the user on the Stripe form
+ * @param  options.headers  The request headers to store on analytics
+ * @return undefined
+ */
+api.payWithStripe = async function payWithStripe (options, stripeInc) {
+  let [
```

Comment:
I think here it would be better to use an object and not an array. this way you're relying on options being passed in a precise order. With the object you can destructure like this: `let {option1, option2, ...} = options`. You may have to enable an additional babel plugin, not sure if our current ones cover this feature